app/models/library_entry.rb

```diff
@@ -164,12 +167,22 @@ def kind
   end
 
   before_save do
-    if status_changed? && completed? && media&.progress_limit
-      # When marked completed, we try to update progress to the cap
-      self.progress = media.progress_limit
-    elsif !status_changed? && progress == media&.progress_limit
-      # When in current and progress equals total episodes
-      self.status = :completed
+    if status_changed? && completed?
+      # update progress to the cap
+      self.progress = media.progress_limit if media&.progress_limit
+      # set finished_at and started_at for the first consume
+      self.started_at ||= Time.now unless imported
+      self.finished_at ||= Time.now unless imported
```

Comment:
Right, but what about a case when the imported data has no values for these, and the `||=` check shouldn't override an already set value since it expands to something like:

```ruby
self.started_at = self.started_at || Time.now
```